### ISSUE TITLE
`dotnet watch run` does not read `appsettings.Development.json` when `--project` is used

### ISSUE THREAD

**Issue by HymanZHAN** [dotnet/sdk#23183]
### Describe the bug

`dotnet watch run` will not read `appsettings.Development.json` of the child project when the `--project` flag is passed and used. It will use the production `appsettings.json` instead.

### To Reproduce

Repro repo: https://github.com/HymanZHAN/DotnetWatchBug
Reproduction instructions are included in the README.

### Exceptions (if any)

```
watch : Started
config token key: 
Unhandled exception. System.ArgumentNullException: String reference not set to an instance of a String. (Parameter 's')
   at System.Text.Encoding.GetBytes(String s) in /builddir/build/BUILD/dotnet-9e8b04bbff820c93c142f99a507a46b976f5c14c/src/runtime.4822e3c3aa77eb82b2fb33c9321f923cf11ddde6/artifacts/source-build/self/src/src/libraries/System.Private.CoreLib/src/System/Text/Encoding.cs:line 665
   at System.Text.UTF8Encoding.UTF8EncodingSealed.GetBytes(String s) in /builddir/build/BUILD/dotnet-9e8b04bbff820c93c142f99a507a46b976f5c14c/src/runtime.4822e3c3aa77eb82b2fb33c9321f923cf11ddde6/artifacts/source-build/self/src/src/libraries/System.Private.CoreLib/src/System/Text/UTF8Encoding.Sealed.cs:line 43
   at API.Extensions.IdentityServiceExtensions.AddIdentityServices(IServiceCollection services, IConfiguration config) in /home/xzhan/Development/Projects/DotnetWatchBug/API/Extensions/IdentityServiceExtension.cs:line 10
   at Program.<Main>$(String[] args) in /home/xzhan/Development/Projects/DotnetWatchBug/API/Program.cs:line 11
watch : Exited with error code 134
watch : Waiting for a file to change before restarting dotnet...
watch : Shutdown requested. Press Ctrl+C again to force exit.
```

### Further technical details
- Include the output of `dotnet --info`

``` shell
.NET SDK (reflecting any global.json):
 Version:   6.0.100
 Commit:    9e8b04bbff

Runtime Environment:
 OS Name:     fedora
 OS Version:  35
 OS Platform: Linux
 RID:         fedora.35-x64
 Base Path:   /usr/lib64/dotnet/sdk/6.0.100/

Host (useful for support):
  Version: 6.0.0
  Commit:  4822e3c3aa

.NET SDKs installed:
  5.0.206 [/usr/lib64/dotnet/sdk]
  6.0.100 [/usr/lib64/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 5.0.9 [/usr/lib64/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.0 [/usr/lib64/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 5.0.9 [/usr/lib64/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.0 [/usr/lib64/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download
```

- The IDE (VS / VS Code/ VS4Mac) you're running on, and its version

```
VS Code 1.63.2
899d46d82c4c95423fb7e10e68eba52050e30ba3
x64
```

**Comment by knumat** [comment]
I am also seeing this issue with dotnet 6.0.101 on Mac OS.

**Comment by knumat** [comment]
To me, this seems like a serious issue since developers might not realize that they are running against the wrong servers, possibly causing data loss or outages on production websites/databases.

**Comment by knumat** [comment]
I found that this this issue only occurs if Hot Reload is enabled.  For example, the following command line works fine for me:
```
dotnet watch --no-hot-reload run --project API
```

If you want to use VS Code with Hot Reload support, update your `.vscode/tasks.json` as follows:
```
        {
            "label": "watch",
            "command": "dotnet",
            "type": "process",
            "args": [
                "watch",
                "run"
            ],
            "options": {
                "cwd": "${workspaceFolder}/API"
            },
            "problemMatcher": "$msCompile"
        }
```

Note that I removed the other command line options due to [this bug](https://github.com/dotnet/aspnetcore/issues/31365) where hot reload is only enabled if there are no arguments.

**Comment by pranavkm** [comment]
Resolved as part of https://github.com/dotnet/sdk/pull/23184. Installer links are currently out of date, but this should be available as part of the 6.0.2xx / 7.0 builds of the .NET SDK installer: https://github.com/dotnet/installer#installers-and-binaries once they are updated.
